.github/workflows/bump.yml

```diff
@@ -3,9 +3,6 @@ permissions:
   contents: read
   pull-requests: write
 on:
-  push:
-    branches:
-      - main
```

Comment:
why do you remove the workflow from the push on `main` branch. It's still needed when we merge in a PR into `main` branch, to deploy the API document to Bump.sh :)

Reply:
With the current setup it was doing all checks twice, and as well as looking untidy there's a carbon footprint to doing everything twice. If you can help me get this right I'd appreciate it.

Reply:
It's *kind of* doing checks twice. 

However, running workflows on the HEAD of a pull request is slightly different than running workflows on the HEAD of the **target branch** (here always `main`).

That's because the merged PR can have difference with the head of the PR (if there's changes on the `main` while the PR is being developed)

Reply:
I don't mean "doing it twice" like once in the PR and once again on main.

I mean it was literally doing it twice on the PR. This happens when you do a first push because both the push event and the pull_request event are triggered, so its literally running all the checks simultaniously twice, once for push and once for pull_request. 

This is a documented thing, and its wasting compute resources: https://github.com/orgs/community/discussions/26940

I've reverted the change here, but I would like to not only fix it in this repo, but in the sample Bump.sh workflow in documentation.

Reply:
Hm :thinking:, I can't see any duplicate jobs on the commits of the HEAD of this PR:

https://github.com/bump-sh-examples/train-travel-api/commit/bc9714035eadf7f51653f5b0bf7c484e5ed5c3e2
![Capture d’écran du 2024-03-06 19-38-43](https://github.com/bump-sh-examples/train-travel-api/assets/904193/7154678d-de85-4737-a8a5-bf64b3ea92c7)

There's only:
- one `Lint` job (the second check `--> Linter OPENAPI` is a created in the `Lint` job by the [super-linter gh action](https://github.com/super-linter/super-linter))
- only one `Check API diff` job.

---

The description of the push target in the GH action workflow file:
```
on:
  push:
    - main
```
Means it will only trigger a workflow on **push** on the **`main` branch** (so not on pushed on any other branches, i.e. pull request branches):

Reply:
The first push will have two lots of checks as you've opened a pull request.

The second push will have only one lot of checks because the pull request is old now.

I'll take a screenshot next time this happens.

Reply:
There's no reason for a duplicate job with the current workflow definition file, sorry to insist!

Here's a demo with a newly created PR with a single push on that PR branch: https://github.com/bump-sh-examples/train-travel-api/pull/13

There's a single `Lint` workflow and a single `Check & Deploy API documentation` workflow` running on the repository:
![Capture d’écran du 2024-03-06 20-15-35](https://github.com/bump-sh-examples/train-travel-api/assets/904193/f43381d6-d086-4f97-bb2c-36fee1cf240c)

Reply:
I saw the duplicates. I saw two of everything. I should have taken a screenshot, but I will do next time it happens.